[fix] sale_by_packaging: update can_be_sold on change

henrybackman opened the pull request:
When changing the packaging type, the `can_be_sold` checkbox did not update on user form to match the selected type

Added onchange function to resolve this.

simahawk commented on sale_by_packaging/models/product_packaging.py:
```diff
@@ -26,6 +26,11 @@ def _compute_can_be_sold(self):
             record.can_be_sold = record.packaging_type_id.can_be_sold
 
+    @api.onchange("packaging_type_id")
+    def _onchange_packaging_type(self):
```
strange... this should be covered by the compute method above :thinking:

sebalix replied:
Why not using a simple related readonly=False + store=True here?
Can't say if it'll solve the issue but I do not understand the current implementation of `_compute_can_be_sold` (unless it is here to be overridden?)

henrybackman committed:
sale_by_packaging: fix can_be_sold update on pkg type change

This shouldn't be required due to the compute method.
However, we experienced some cases where nested form inside the product
form fails to update the flag without this onchange.

ivantodorovich reviewed (approved):
weird, indeed..

henrybackman commented:
After further investigation, this change is not anymore needed. There was a missing `@api.depends()` on a compute function override that was causing the values not to update on the form. Closing this PR.
